icu4c/source/common/localebuilder.cpp

```diff
@@ -167,14 +166,17 @@ _copyExtensions(const Locale& from, Locale* to, bool validate, UErrorCode& error
         CharString value;
         CharStringByteSink sink(&value);
         from.getKeywordValue(key, sink, errorCode);
+        if (U_FAILURE(errorCode)) { return; }
         if (uprv_strcmp(key, kAttributeKey) == 0) {
             transform(value.data(), value.length());
         }
         if (validate &&
             !_isKeywordValue(key, value.data(), value.length())) {
             errorCode = U_ILLEGAL_ARGUMENT_ERROR;
+            return;
         }
         to->setKeywordValue(key, value.data(), errorCode);
+        if (U_FAILURE(errorCode)) { return; }
```

Comment:
Optional: Ok, but should not be necessary -- iter->next() should return nullptr if failure.